comparative_benchmark/benchmark_lib.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2023 The OpenXLA Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import argparse
+import dataclasses
+import json
+import multiprocessing
+import pathlib
+import re
+import sys
+from typing import Any, Callable, Dict, List, Sequence
+
+# Add common_benchmark_suite dir to the search path.
+sys.path.insert(
+    0, str(pathlib.Path(__file__).parents[2] / "common_benchmark_suite"))
+# Add comparative_benchmark dir to the search path.
+sys.path.insert(
+    0, str(pathlib.Path(__file__).parents[2] / "comparative_benchmark"))
+
+from openxla.benchmark import def_types
+import utils
+
+
+def _run(benchmark: def_types.BenchmarkCase, run_in_process: bool,
+         warmup_iterations: int, iterations: int,
+         input_npys: Sequence[pathlib.Path],
+         expect_npys: Sequence[pathlib.Path],
+         benchmark_function: Callable) -> utils.BenchmarkResult:
+  model = benchmark.model
+  input_data = benchmark.input_data.artifacts[
+      def_types.ModelTestDataFormat.NUMPY_TENSORS]
+  expected_output = benchmark.expected_output.artifacts[
+      def_types.ModelTestDataFormat.NUMPY_TENSORS]
+
+  data_type = model.model_parameters["data_type"]
+  batch_size = model.model_parameters["batch_size"]
+  input_dims = input_data.data_parameters["tensor_dimensions"]
+  output_dims = expected_output.data_parameters["tensor_dimensions"]
+  benchmark_definition = {
+      "benchmark_id": benchmark.id,
+      "benchmark_name": benchmark.name,
+      "framework": str(model.model_impl.framework_type),
+      "data_type": data_type,
+      "batch_size": batch_size,
+      "inputs": input_dims,
+      "outputs": output_dims,
+      "compiler": "xla",
```

Comment:
This should be configurable?

Reply:
Let's make it configurable once we add another compiler.